code/game/machinery/doors/firedoor.dm

```diff
@@ -139,6 +141,30 @@
 	else
 		close()
 
+/// A simple toggle for firedoors between on and off
+/obj/machinery/door/firedoor/try_to_crowbar_secondary(obj/item/acting_object, mob/user)
+	if(welded || operating)
+		return
+
+	if(density)
+		being_held_open = TRUE
+		open()
+		RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/handle_held_open_adjacency)
+		RegisterSignal(user, COMSIG_LIVING_SET_BODY_POSITION, .proc/handle_held_open_adjacency)
+		handle_held_open_adjacency(user)
+	else
+		close()
+
+/obj/machinery/door/firedoor/proc/handle_held_open_adjacency(mob/user)
+	SIGNAL_HANDLER
+	var/mob/living/living_user = user
+	if(Adjacent(user) && isliving(user) && (living_user.body_position == STANDING_UP))
+		return
+	being_held_open = FALSE
+	addtimer(CALLBACK(src, .proc/close), 0.1 SECONDS)
```

Comment:
What's the goal of a timer with such a short length? Are you sure you didn't want `INVOKE_ASYNC`?

Reply:
I didn't know that that thing exists, I'll change it right away. Thanks!